docker/DockerfileNoetic

```diff
@@ -0,0 +1,77 @@
+# Start with ROS base image
+FROM ros:noetic-ros-base
+
+# Tools useful during development.
+RUN apt update \
+ && apt install -y --no-install-recommends\
+        cppcheck \
```

Comment:
```suggestion
        build-essential \
        cppcheck \
```
Also, we like to use alphanumeric ordering for long lists like this, to make items easy to find.

Reply:
I reorganized to the recommend layout in bc9997dbfa - much clearer